mimesis/builtins/ru.py

```diff
@@ -90,16 +91,38 @@ def series_and_number(self) -> str:
     def snils(self) -> str:
         """Generate invalid ``SNILS``.
 
-        This method does not generate SNILS using algorithm and
-        it's mean that SNILS generated by this method can be invalid.
-
         :return: SNILS.
 
         :Example:
             451-952-540-41.
         """
-        mask = '###-###-###-##'
-        return self.random.custom_code(mask=mask)
+        snils = [self.random.randint(0, 9) for _ in range(9)]
+        while int(reduce(lambda x, y: x + y,
+                         map(lambda x: str(x), snils))) <= 1001998:
+            snils = [self.random.randint(0, 9) for _ in range(9)]
+
+        def get_snils_sum(snils):
+            s = 0
+            for index, mult in enumerate(range(9, 0, -1)):
+                s += snils[index] * mult
+
+            def ret_s(s):
```

Comment:
There are several things with this function that could be improved:
1. `ret_s` is not a very descriptive name, it should be enhanced 
2. `ret_s` has a nested level of 3. That's way to much!